This patch changes the default user under which quagga daemons run to daemon.
This will be submitted to the community version and this patch may be removed
when upgrading to a version of Quagga with this fix already applied. 

--- solaris/quagga.xml.in
+++ solaris/quagga.xml.in
@@ -65,8 +65,9 @@
 		timeout_seconds='60'>
 		<method_context>
 		  <method_credential
-		   user='root' group='root'
-                   privileges='all,{zone}:/system/volatile/quagga,{zone}:/system/volatile/quagga/*' />
+		   user='root' group='root'
+                   privileges=
+'basic,net_icmpaccess,net_rawaccess,sys_admin,sys_ip_config,net_privaddr,proc_setid,sys_net_config,sys_dl_config,proc_chroot,proc_priocntl,file_owner,file_dac_execute,file_dac_read,file_dac_write,file_dac_search,{zone}:/etc/quagga/*,{zone}:/system/volatile/quagga,{zone}:/system/volatile/quagga/*' />
  		</method_context>
 	</exec_method>
 
@@ -196,8 +197,9 @@
 		timeout_seconds='60'>
 		<method_context>
 		  <method_credential
-		   user='root' group='root'
-                   privileges='all,{zone}:/system/volatile/quagga,{zone}:/system/volatile/quagga/*' />
+		   user='root' group='root'
+                   privileges=
+'basic,net_icmpaccess,net_rawaccess,sys_admin,sys_ip_config,net_privaddr,proc_setid,sys_net_config,sys_dl_config,proc_chroot,proc_priocntl,file_owner,file_dac_execute,file_dac_read,file_dac_write,file_dac_search,{zone}:/etc/quagga/*,{zone}:/system/volatile/quagga,{zone}:/system/volatile/quagga/*' />
  		</method_context>
 	</exec_method>
 
@@ -318,8 +320,9 @@
 		timeout_seconds='60'>
 		<method_context>
 		  <method_credential
-		   user='root' group='root'
-                   privileges='all,{zone}:/system/volatile/quagga,{zone}:/system/volatile/quagga/*' />
+		   user='root' group='root'
+                   privileges=
+'basic,net_icmpaccess,net_rawaccess,sys_admin,sys_ip_config,net_privaddr,proc_setid,sys_net_config,sys_dl_config,proc_chroot,proc_priocntl,file_owner,file_dac_execute,file_dac_read,file_dac_write,file_dac_search,{zone}:/etc/quagga/*,{zone}:/system/volatile/quagga,{zone}:/system/volatile/quagga/*' />
  		</method_context>
 	</exec_method>
 
@@ -440,8 +443,9 @@
 		timeout_seconds='60'>
 		<method_context>
 		  <method_credential
-		   user='root' group='root'
-                   privileges='all,{zone}:/system/volatile/quagga,{zone}:/system/volatile/quagga/*' />
+		   user='root' group='root'
+                   privileges=
+'basic,net_icmpaccess,net_rawaccess,sys_admin,sys_ip_config,net_privaddr,proc_setid,sys_net_config,sys_dl_config,proc_chroot,proc_priocntl,file_owner,file_dac_execute,file_dac_read,file_dac_write,file_dac_search,{zone}:/etc/quagga/*,{zone}:/system/volatile/quagga,{zone}:/system/volatile/quagga/*' />
  		</method_context>
 	</exec_method>
 
@@ -565,8 +569,9 @@
 		timeout_seconds='60'>
 		<method_context>
 		  <method_credential
-		   user='root' group='root'
-                   privileges='all,{zone}:/system/volatile/quagga,{zone}:/system/volatile/quagga/*' />
+		   user='root' group='root'
+                   privileges=
+'basic,net_icmpaccess,net_rawaccess,sys_admin,sys_ip_config,net_privaddr,proc_setid,sys_net_config,sys_dl_config,proc_chroot,proc_priocntl,file_owner,file_dac_execute,file_dac_read,file_dac_write,file_dac_search,{zone}:/etc/quagga/*,{zone}:/system/volatile/quagga,{zone}:/system/volatile/quagga/*' />
  		</method_context>
 	</exec_method>
 
@@ -685,8 +690,9 @@
 		timeout_seconds='60'>
 		<method_context>
 		  <method_credential
-		   user='root' group='root'
-                   privileges='all,{zone}:/system/volatile/quagga,{zone}:/system/volatile/quagga/*' />
+		   user='root' group='root'
+                   privileges=
+'basic,net_icmpaccess,net_rawaccess,sys_admin,sys_ip_config,net_privaddr,proc_setid,sys_net_config,sys_dl_config,proc_chroot,proc_priocntl,file_owner,file_dac_execute,file_dac_read,file_dac_write,file_dac_search,{zone}:/etc/quagga/*,{zone}:/system/volatile/quagga,{zone}:/system/volatile/quagga/*' />
  		</method_context>
 	</exec_method>
 
@@ -820,8 +826,9 @@
 		timeout_seconds='60'>
 		<method_context>
 		  <method_credential
-		   user='root' group='root'
-                   privileges='all,{zone}:/system/volatile/quagga,{zone}:/system/volatile/quagga/*' />
+		   user='root' group='root'
+                   privileges=
+'basic,net_icmpaccess,net_rawaccess,sys_admin,sys_ip_config,net_privaddr,proc_setid,sys_net_config,sys_dl_config,proc_chroot,proc_priocntl,file_owner,file_dac_execute,file_dac_read,file_dac_write,file_dac_search,{zone}:/etc/quagga/*,{zone}:/system/volatile/quagga,{zone}:/system/volatile/quagga/*' />
  		</method_context>
 	</exec_method>
 
